### PR TITLE
Add support for dashboard, some scheduled searches, and recovery for a single folder only

### PR DESCRIPTION
**Pull request by emichaeli**

### CODE REVIEW

File: src/search.py [hunk 1]
```diff
@@ -78,19 +83,161 @@ def getAutoComplete():
...
+        return {}
+
+
+# Converts an integer to 16-digit hex (prepending 0's if needed)
+def convertToHex(number):
+    hexString = hex(number).lstrip("0x")
+    hexLength = len(hexString)
+    if hexLength < 16:
+        hexString = ("0"*(16-hexLength)) + hexString
+    return hexString
```
Comment (sumovishal) [inline src/search.py:131]: similar comment here for hex conversion.

**Review by sumovishal — APPROVED** [review]
some minor comments, rest lgtm.

File: src/dashboard.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+def convertToHex(number):
+    if number < 0:
+        offset = 1 << 64
+        mask = offset - 1
+        number = number + offset & mask
+    hexString = hex(number).lstrip("0x")
+    hexLength = len(hexString)
+    if hexLength < 16:
+        hexString = ("0"*(16-hexLength)) + hexString
+    return hexString
```
[the remark this answers is not in the document]
Comment (emichaeli) [reply src/dashboard.py]: nice :)

File: src/dashboard.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        self.title = row[1].decode("utf-8")
+        self.panelType = row[2]
+        self.visualSettings = row[3].decode("utf-8")
+        self.timeRange = row[4]
+        self.queries = []
+
+    def asdict(self):
+        return {
+            'key': self.key,
+            'title': self.title.replace(u"\u00A0", " "),
```
[the remark this answers is not in the document]
Comment (emichaeli) [reply src/dashboard.py:49]: yeah that was a bit annoying, ran into a bunch of encoding issues when using the data from DB in API. For some reason the API failed when `\u00A0` character was present, and replacing it with spaces fixed it

[the remark this answers is not in the document]
Comment (sumovishal) [reply src/dashboard.py:49]: ok, let's leave a comment.

[the remark this answers is not in the document]
Comment (emichaeli) [reply src/dashboard.py:50]: good point, added

**Pull request merged**